gui/widgets/editable.cpp

```diff
@@ -262,8 +316,25 @@ bool EditableWidget::handleKeyDown(Common::KeyState state) {
 }
 
 void EditableWidget::defaultKeyDownHandler(Common::KeyState &state, bool &dirty, bool &forcecaret, bool &handled) {
-	if (state.ascii < 256 && tryInsertChar((byte)state.ascii, _caretPos)) {
+	//  if highlighted, replace selected text with inserted character
+	if (_highlightVisible) {
+		if (state.ascii < 256 && tryInsertChar((byte)state.ascii, _caretPos)) {
+			for (uint32 i = _highlightPos + _highlightCharacterCount; i > (uint32)_highlightPos; i--) {
+				_editString.deleteChar(i - 1);
```

Comment:
please use `Common::String::erase(uint32 pos, uint32 len)` instead of this loop.